From 215ba2b9052292d35d7f207dfcdc8639920bcfaf Mon Sep 17 00:00:00 2001
From: Andreas Tille <tille@debian.org>
Date: Sun, 26 Aug 2012 23:13:29 +0200
Subject: [PATCH] Perls move() function is unreliable when moving directories
 so we need to check in advance and prevent using move()
 when the source is a directory

---
 scripts/uscan.pl |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scripts/uscan.pl b/scripts/uscan.pl
index f575a0f..f6046b0 100755
--- a/scripts/uscan.pl
+++ b/scripts/uscan.pl
@@ -2114,11 +2114,11 @@ sub get_main_source_dir($$$$) {
     mkdir($main_source_dir) or die("Unable to create temporary source directory $main_source_dir\n");
     foreach my $file (@files) {
 	unless ($file =~ /^\.\.?/) {
-            # move("${tempdir}/$file", $main_source_dir) or die("Unable to move ${tempdir}/$file directory $main_source_dir\n");
-            unless ( move("${tempdir}/$file", $main_source_dir) ) {
-            # HELP: why can't perl move not move directories????
-                print "Perl move seems to be not able to ` move(\"${tempdir}/$file\", $main_source_dir) ` ... trying system mv\n" if $debug;
+	    if ( -d "${tempdir}/$file" ) {
+                # HELP: why can't perl move not move directories????
                 system( "mv ${tempdir}/$file $main_source_dir" ) ;
+            } else {
+                move("${tempdir}/$file", $main_source_dir) or die("Unable to move ${tempdir}/$file directory $main_source_dir\n");
             }
         }
     }
-- 
1.7.10.4

